src/documentmanager/documentmanager.cpp

```diff
@@ -325,43 +333,39 @@ QMdiArea* DocumentManager::lastAreaInFocus()
     // search for area which accomodates last doc in focus
     for (const auto& area: mDocAreas)
     {
-        auto subWdwList = area->subWindowList();
-
-        areaInFocusIter = std::find_if(subWdwList.cbegin(), subWdwList.cend(),
-                                       [this](const auto& doc)
-        {
-            return static_cast<CodeEditor*>(doc->widget()) == mpPrevEditorInFocus;
-        });
+        auto currentWdw = area->currentSubWindow();
 
-        if (areaInFocusIter != subWdwList.end())
+        if (currentWdw)
         {
-            return area;
+            auto doc = qobject_cast<CodeEditor*>(currentWdw->widget());
+            if (doc && doc == mpPrevEditorInFocus)
```

Comment:
`if (doc && doc == mpPrevEditorInFocus)`
in accordance to priority of operations, rewrite it to
`if (doc && (doc == mpPrevEditorInFocus))`

Reply:
fixed